app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,338 @@
+        description = description,
+        onClick = onClick,
+        afterListAction = {
+            endIconPainter?.let {
+                ListItemIcon(
+                    painter = endIconPainter,
+                    iconDescription = endIconDescription,
+                    onClick = onEndIconClick,
+                )
+            }
+        }
+    )
+}
+
+/**
+ * List item used to display a label and a [Favicon] with an optional description text and
+ * an optional, interactable icon at the end.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param url Website [url] for which the favicon will be shown.
+ * @param endIconPainter [Painter] used to display a [ListItemIcon] at the end of the list item.
+ * @param endIconDescription Content description of the end icon.
+ * @param onEndIconClick Called when the user clicks on the end icon.
+ */
+@Composable
+fun FaviconListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    url: String,
+    endIconPainter: Painter? = null,
+    endIconDescription: String? = null,
+    onEndIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        beforeListAction = {
+            Spacer(modifier = Modifier.width(16.dp))
+
+            Favicon(
+                url = url,
+                size = ICON_SIZE,
+            )
+
+            Spacer(modifier = Modifier.width(16.dp))
+        },
+        afterListAction = {
+            endIconPainter?.let {
+                ListItemIcon(
+                    painter = endIconPainter,
+                    iconDescription = endIconDescription,
+                    onClick = onEndIconClick,
+                )
+            }
+        }
+    )
+}
+
+/**
+ * List item used to display a label and an icon at the beginning with an optional description
+ * text and an optional, interactable icon at the end.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param startIconPainter [Painter] used to display a [ListItemIcon] at the start of the list item.
+ * @param startIconDescription Content description of the drawable.
+ * @param endIconPainter [Painter] used to display a [ListItemIcon] at the end of the list item.
+ * @param endIconDescription Content description of the end icon.
+ * @param onEndIconClick Called when the user clicks on the end icon.
+ */
+@Composable
+fun IconListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    startIconPainter: Painter,
+    startIconDescription: String? = null,
+    endIconPainter: Painter? = null,
+    endIconDescription: String? = null,
+    onEndIconClick: (() -> Unit)? = null,
```

Comment:
We are kinda mixing start/end with before/after here. I think we should just start with before/after.

Reply:
Agreed. Updated